test/extensions/filters/common/local_ratelimit/local_ratelimit_test.cc

```diff
@@ -394,6 +430,44 @@ TEST_F(LocalRateLimiterDescriptorImplTest, TokenBucketDifferentDescriptorDiffere
   EXPECT_FALSE(rate_limiter_->requestAllowed(descriptor_));
 }
 
+// Verify token bucket status of max tokens, remaining tokens and remaining fill interval.
+TEST_F(LocalRateLimiterDescriptorImplTest, TokenBucketDescriptorStatus) {
+  TestUtility::loadFromYaml(fmt::format(single_descriptor_config_yaml, 2, 2, "3s"),
+                            *descriptors_.Add());
+  initializeWithDescriptor(std::chrono::milliseconds(3000), 2, 2);
+
+  /// 2 -> 1 tokens
+  EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(3000), nullptr));
+  EXPECT_TRUE(rate_limiter_->requestAllowed(descriptor_));
+  EXPECT_EQ(rate_limiter_->maxTokens(descriptor_), 2);
+  EXPECT_EQ(rate_limiter_->remainingTokens(descriptor_), 1);
+  EXPECT_EQ(rate_limiter_->remainingFillInterval(descriptor_), 3);
+
+  // 1 -> 0 tokens
+  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(1000), dispatcher_,
+                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
+  EXPECT_TRUE(rate_limiter_->requestAllowed(descriptor_));
+  EXPECT_EQ(rate_limiter_->maxTokens(descriptor_), 2);
+  EXPECT_EQ(rate_limiter_->remainingTokens(descriptor_), 0);
+  EXPECT_EQ(rate_limiter_->remainingFillInterval(descriptor_), 2);
+
+  // 0 -> 0 tokens
+  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(1000), dispatcher_,
+                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
+  EXPECT_FALSE(rate_limiter_->requestAllowed(descriptor_));
+  EXPECT_EQ(rate_limiter_->maxTokens(descriptor_), 2);
+  EXPECT_EQ(rate_limiter_->remainingTokens(descriptor_), 0);
+  EXPECT_EQ(rate_limiter_->remainingFillInterval(descriptor_), 1);
+
+  // 0 -> 2 tokens
+  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(1000), dispatcher_,
+                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
+  fill_timer_->invokeCallback();
+  EXPECT_EQ(rate_limiter_->maxTokens(descriptor_), 2);
+  EXPECT_EQ(rate_limiter_->remainingTokens(descriptor_), 2);
+  EXPECT_EQ(rate_limiter_->remainingFillInterval(descriptor_), 3);
+}
+
```

Comment:
I think It is be a good idea to add a unit test to verify the functionality with multiple descriptors (`multiple_descriptor_config_yaml`) as well. WDYT?

Reply:
Yes, I will add a unit test for it.